File: src/java/TastyFishBurgerAdapter.java
package com.logview.tastyfishburger;

import com.logview.core.LogViewMachine;
import com.logview.core.LogEntry;
import com.logview.core.LogFilters;
import java.time.Instant;
import java.util.List;
import java.util.Map;

public class TastyFishBurgerAdapter {
    private final LogViewMachine machine;

    public TastyFishBurgerAdapter(LogViewMachine machine) {
        this.machine = machine;
    }

    public List<LogEntry> fetchLogs(LogFilters filters) {
        return machine.fetchLogs(filters);
    }

    public List<LogEntry> updateFilters(LogFilters newFilters) {
        return machine.updateFilters(newFilters);
    }

    public void addLog(String level, String message, Map<String, Object> metadata) {
        LogEntry entry = new LogEntry(level, message, metadata);
        machine.addLog(entry);
    }

    public void clearLogs() {
        machine.clearLogs();
    }

    public static class FishBurgerLog {
        private final String id;
        private final Instant timestamp;
        private final String level;
        private final String message;
        private final Map<String, Object> metadata;

        public FishBurgerLog(String id, Instant timestamp, String level, String message, Map<String, Object> metadata) {
            this.id = id;
            this.timestamp = timestamp;
            this.level = level;
            this.message = message;
            this.metadata = metadata;
        }

        public String getId() { return id; }
        public Instant getTimestamp() { return timestamp; }
        public String getLevel() { return level; }
        public String getMessage() { return message; }
        public Map<String, Object> getMetadata() { return metadata; }
    }

    public FishBurgerLog convertToFishBurgerLog(LogEntry entry) {
        return new FishBurgerLog(
            entry.getId(),
            entry.getTimestamp(),
            entry.getLevel(),
            entry.getMessage(),
            entry.getMetadata()
        );
    }
} 

File: example/src/main/kotlin/com/logview/example/tastyfishburger/TastyFishBurgerAdapter.kt
package com.logview.example.tastyfishburger

import com.logview.core.LogViewMachine
import com.logview.core.LogEntry
import com.logview.core.LogFilter
import com.logview.core.ViewStateMachine
import java.time.Instant

class TastyFishBurgerAdapter(
    private val stateMachine: TastyFishBurgerStateMachine
) {
    private val logViewMachine = stateMachine.getLogViewMachine()
    private val viewStateMachine = stateMachine.getViewStateMachine()

    fun fetchLogs(filter: LogFilter? = null): List<LogEntry> {
        return logViewMachine.getLogs(filter)
    }

    fun updateFilters(newFilter: LogFilter): List<LogEntry> {
        return logViewMachine.getLogs(newFilter)
    }

    fun addLog(level: String, message: String, metadata: Map<String, Any>? = null) {
        logViewMachine.addLog(LogEntry(
            id = "",
            timestamp = Instant.EPOCH,
            level = level,
            message = message,
            metadata = metadata
        ))
    }

    fun clearLogs() {
        logViewMachine.clearLogs()
    }

    fun getViewStateMachineLogs(): List<ViewStateMachine.LogEntry> {
        return viewStateMachine.getLogEntries()
    }

    fun subscribeToLogs(callback: (List<LogEntry>) -> Unit): () -> Unit {
        return logViewMachine.subscribe(callback)
    }

    data class FishBurgerLog(
        val id: String,
        val timestamp: Instant,
        val level: String,
        val message: String,
        val metadata: Map<String, Any>? = null
    )

    fun convertToFishBurgerLog(entry: LogEntry): FishBurgerLog {
        return FishBurgerLog(
            id = entry.id,
            timestamp = entry.timestamp,
            level = entry.level,
            message = entry.message,
            metadata = entry.metadata
        )
    }

    // Enhanced methods for ViewStateMachine integration
    suspend fun executeState(stateName: String, data: TastyFishBurgerStateMachine.FishBurgerData) {
        viewStateMachine.executeState(stateName, data)
    }

    fun getStateMachineLogs(): List<ViewStateMachine.LogEntry> {
        return viewStateMachine.getLogEntries()
    }

    fun clearStateMachineLogs() {
        viewStateMachine.clearLogs()
    }
} 